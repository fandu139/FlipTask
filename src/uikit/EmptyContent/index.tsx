import AppStyles from '../../theme/appStyles';
import React from 'react';
import { View, StyleSheet, Text, Image } from 'react-native';
import { IMAGE_DEFAULT_EMPTY_LIST } from '../../assets/image';

type Props = {
  image?: number;
  title?: string;
  testID?: string;
  accessibilityLabel?: string;
};

const EmptyContent: React.FC<Props> = ({ title, image, testID, accessibilityLabel }: Props) => (
  <View
    style={AppStyles.fillCenterContent}
    testID={testID}
    accessibilityLabel={accessibilityLabel || testID}
  >
    <Image
      source={image}
      style={styles.image}
      testID={testID}
      accessibilityLabel={accessibilityLabel}
    />
    <Text>{title}</Text>
  </View>
);

const styles = StyleSheet.create({
  image: {
    width: 200,
    height: 200,
    marginBottom: 20,
  },
});

EmptyContent.defaultProps = {
  image: IMAGE_DEFAULT_EMPTY_LIST,
  title: 'Data tidak ditemukan',
  testID: 'view-empty-content',
  accessibilityLabel: 'view-empty-content',
};

export default EmptyContent;
