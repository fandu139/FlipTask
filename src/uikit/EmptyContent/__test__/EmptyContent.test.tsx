import EmptyContent from '../../EmptyContent';
import React from 'react';
import { render } from '@testing-library/react-native';

describe('empty content component test', () => {
  describe('snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<EmptyContent />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
