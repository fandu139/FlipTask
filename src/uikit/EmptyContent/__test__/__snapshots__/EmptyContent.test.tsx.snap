// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty content component test snapshot test should render correctly 1`] = `
<View
  accessibilityLabel="view-empty-content"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="view-empty-content"
>
  <Image
    accessibilityLabel="view-empty-content"
    source="test-file-stub"
    style={
      Object {
        "height": 200,
        "marginBottom": 20,
        "width": 200,
      }
    }
    testID="view-empty-content"
  />
  <Text>
    Data tidak ditemukan
  </Text>
</View>
`;
